eng/common/scripts/git-branch-push.ps1

```diff
@@ -41,27 +42,38 @@ param(
 # would fail the first time git wrote command output.
 $ErrorActionPreference = "Continue"
 
-if (!(git remote | ? {$_ -eq $RemoteName}))
+if ((git remote) -contains $RemoteName)
 {
-    Write-Host "git remote add $RemoteName $GitUrl"
-    git remote add $RemoteName $GitUrl
-    if ($LASTEXITCODE -ne 0)
-    {
-        Write-Error "Unable to add remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
-        exit $LASTEXITCODE
-    }
+  Write-Host "git remote get-url $RemoteName"
+  $remoteUrl = git remote get-url $RemoteName
+  if ($remoteUrl -ne $GitUrl)
+  {
+    Write-Error "Remote with name $RemoteName already exists with an incompatible url [$remoteUrl] which should be [$GitUrl]."
+    exit 1
+  }
 }
-
-Write-Host "git fetch $RemoteName"
-git fetch $RemoteName
-if ($LASTEXITCODE -ne 0)
+else 
 {
-    Write-Error "Unable to fetch remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
+  Write-Host "git remote add $RemoteName $GitUrl"
+  git remote add $RemoteName $GitUrl
+  if ($LASTEXITCODE -ne 0)
+  {
+    Write-Error "Unable to add remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
     exit $LASTEXITCODE
+  }
+}
+# Checkout to PRBranch, create new one if not exists.
+try 
+{
+    Write-Host "git checkout -b $PRBranchName"
+    git checkout -b $PRBranchName
```

Comment:
I don't believe this will throw an exception you probably need to capture the LASTEXITCODE like other cases.

Reply:
Right. It is not throwing exception.

`(git checkout $PRBranchName | git checkout -b $PRBranchName) 2>&1`
Can we do this and set the `$LASTEXITCODE = 0`?